frontend/dockerfile/dockerfile2llb/convert.go

```diff
@@ -953,6 +962,7 @@ func dispatch(d *dispatchState, cmd command, opt dispatchOpt) error {
 			params:          c.SourcesAndDest,
 			excludePatterns: c.ExcludePatterns,
 			source:          l,
+			fromStage:       c.From,
```

Comment:
maybe something like `ignoredPatterns []string` and if `if len(cmd.sources) != 0 {` (line 954) then `ignoredPatterns = opt.dockerIgnorePatterns`